packages/protocol/src/action-protocol/contexts.ts

```diff
@@ -14,45 +14,83 @@
+import { hasArrayProp, hasObjectProp, hasStringProp } from '../utils/type-util';
+import { Action, RequestAction, ResponseAction } from './base-protocol';
 import { Args, EditorContext, LabeledAction } from './types';
 
 /**
  * The `RequestContextActions` is sent from the client to the server to request the available actions for the context with id contextId.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `RequestContextActions`.
  */
-export class RequestContextActions implements RequestAction<SetContextActions> {
-    static readonly KIND = 'requestContextActions';
-    constructor(
-        public readonly contextId: string,
-        public readonly editorContext: EditorContext,
-        public readonly requestId: string = generateRequestId(),
-        public readonly kind: string = RequestContextActions.KIND
-    ) {}
+export interface RequestContextActions extends RequestAction<SetContextActions> {
+    /**
+     * The unique action kind.
+     */
+    kind: typeof RequestContextActions.KIND;
+
+    /**
+     * The identifier for the context.
+     */
+    contextId: string;
+
+    /**
+     * The current editor context.
+     */
+    editorContext: EditorContext;
 }
 
-export function isRequestContextActions(action: any): action is RequestContextActions {
-    return (
-        isActionKind(action, RequestContextActions.KIND) &&
-        isString(action, 'contextId') &&
-        isObject(action, 'editorContext') &&
-        isString(action, 'requestId')
-    );
+export namespace RequestContextActions {
+    export const KIND = 'requestContextActions';
+
+    export function is(object: any): object is RequestContextActions {
+        return RequestAction.hasKind(object, KIND) && hasStringProp(object, 'contextId') && hasObjectProp(object, 'editorContext');
+    }
+
+    export function create(options: { contextId: string; editorContext: EditorContext; requestId?: string }): RequestContextActions {
+        return {
+            kind: KIND,
+            requestId: '',
+            ...options
+        };
+    }
 }
 
 /**
  * The `SetContextActions` is the response to a {@link RequestContextActions} containing all actions for the queried context.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `SetContextsActions`.
  */
-export class SetContextActions implements ResponseAction {
-    static readonly KIND = 'setContextActions';
-    constructor(
-        public readonly actions: LabeledAction[],
-        public readonly responseId: string = '',
-        public readonly args?: Args,
-        public readonly kind: string = SetContextActions.KIND
-    ) {}
+export interface SetContextActions extends ResponseAction {
+    /**
+     * The unique action kind.
+     */
+    kind: typeof SetContextActions.KIND;
```

Comment:
I'd spare such comments, also if it hurts to be "incomplete". But imho they just add noise and add no information.